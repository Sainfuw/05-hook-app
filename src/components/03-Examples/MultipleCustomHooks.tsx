import React from "react";
import "../02-useEffect/effects.css";
import { useFetch, QuoteInterface } from "../../hooks/useFetch";
import { useCounter } from "../../hooks/useCounter";

export const MultipleCustomHooks = () => {
  const { counter, increment, decrement } = useCounter(1);
  const api: QuoteInterface = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );
  const { data, loading } = api;

  return (
    <>
      <h4>BreakingBad Quotes</h4>
      <hr />

      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-end">
          <p className="mb-2">{data!.quote}</p>
          <footer className="blockquote-footer">{data!.author}</footer>
        </blockquote>
      )}
      <div className="text-center">
        <button onClick={decrement} className="btn btn-primary mx-2">
          Prev Quote
        </button>
        <button onClick={increment} className="btn btn-primary mx-2">
          Next Quote
        </button>
      </div>
    </>
  );
};
