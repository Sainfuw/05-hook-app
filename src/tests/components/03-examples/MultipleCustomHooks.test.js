import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-Examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en MultipleCustomHooks.test", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
      decrement: () => {},
    });
  });

  test("debe mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: {},
      loading: true,
      error: "",
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar la informacion", () => {
    useFetch.mockReturnValue({
      data: {
        author: "Emilio",
        quote: "Hola Mundo",
      },
      loading: false,
      error: "",
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-2").text().trim()).toBe("Hola Mundo");
    expect(wrapper.find("footer").text().trim()).toBe("Emilio");
  });
});
