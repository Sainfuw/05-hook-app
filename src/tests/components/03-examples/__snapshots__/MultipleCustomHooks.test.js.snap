// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHooks.test debe mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    BreakingBad Quotes
  </h4>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <div
    className="text-center"
  >
    <button
      className="btn btn-primary mx-2"
      onClick={[Function]}
    >
      Prev Quote
    </button>
    <button
      className="btn btn-primary mx-2"
      onClick={[Function]}
    >
      Next Quote
    </button>
  </div>
</Fragment>
`;
